lib/chef/resource/user.rb

```diff
@@ -67,10 +67,12 @@ class User < Chef::Resource
         default: false
 
       property :uid, [ String, Integer, NilClass ], # nil for backwards compat
-        description: "The numeric user identifier."
+        description: "The numeric user identifier.",
+        coerce: proc { |n| n && Integer(n) }
 
       property :gid, [ String, Integer, NilClass ], # nil for backwards compat
-        description: "The numeric group identifier."
+        description: "The numeric group identifier.",
+        coerce: proc { |n| n && Integer(n) }
```

Comment:
This allows setting uid and gid to nil or something coercible to Integer. I could preserve the existing error behavior by changing this to `coerce: proc { |n| n && Integer(n) rescue n }`, but there are many other places in resources where a `coerce: proc { |n| Integer(n) }` is used.